apps/src/aichat/views/ChatMessageView.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {useMemo, useState} from 'react';
+
+import ChatMessage from '@cdo/apps/aiComponentLibrary/chatMessage/ChatMessage';
+import {Role} from '@cdo/apps/aiComponentLibrary/chatMessage/types';
+import {getChatMessageDisplayText} from '@cdo/apps/aiComponentLibrary/chatMessage/utils';
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import {AiInteractionStatus as Status} from '@cdo/generated-scripts/sharedConstants';
+
+import aichatI18n from '../locale';
+import {ChatMessage as ChatMessageType} from '../types';
+
+import TeacherFeedbackFooter from './TeacherFeedbackFooter';
+
+import moduleStyles from '@cdo/apps/aiComponentLibrary/chatMessage/chat-message.module.scss';
+
+interface ChatMessageViewProps {
+  chatMessage: ChatMessageType;
+  isChatHistoryView: boolean;
+}
+
+const ChatMessageView: React.FunctionComponent<ChatMessageViewProps> = ({
+  chatMessage,
+  isChatHistoryView,
+}) => {
+  const [showProfaneUserMessage, setShowProfaneUserMessage] = useState(false);
+
+  const getDisplayText: string = useMemo(() => {
+    return getChatMessageDisplayText(
+      chatMessage.status,
+      chatMessage.role,
+      chatMessage.chatMessageText,
+      showProfaneUserMessage
+    );
+  }, [chatMessage, showProfaneUserMessage]);
+
+  return (
+    <ChatMessage
+      {...chatMessage}
+      showProfaneUserMessage={showProfaneUserMessage}
+    >
+      {isChatHistoryView &&
+        getDisplayText === chatMessage.chatMessageText &&
```

Comment:
For my understanding, what's the extra check for here? Could we just check for `status === Status.OK`?

Reply:
There's a bunch of logic in `getChatMessageDisplayText` that sometimes returns `chatMessageText` and sometimes a pre-canned string depending on the `Status` and `Role` and regardless of what that logic is, it only makes sense to give the teacher a flag to click if they can see what the message text actually says so I thought that was a more direct check.